grr/server/grr_response_server/gui/ui/lib/api_translation/client.ts

```diff
@@ -14,6 +14,15 @@ function createKnowledgeBase(kb: ApiKnowledgeBase): KnowledgeBase {
   };
 }
 
+export function translateClientLabel(label: ApiClientLabel): ClientLabel {
+  let owner = '';
+
+  if (!label.name) throw new Error('name attribute is missing.');
+  if (label.owner) owner = label.owner;
```

Comment:
The purpose of functions in api_translation is to convert API data structures that are heavily influenced by the Protobuf language to pure, plain-old TypeScript objects. In Proto best practices, all fields should be `optional`, because `required` fields can NEVER be changed to `optional` in the future.

Assigning the empty string to `label.owner` is a hack to satisfy the type system, but makes using ClientLabel hard. With an empty string as replacement for `undefined`, code now needs to check for the special value of `''` to really be sure that the ClientLabel has an owner.

Please instead, throw an Error if label.owner is false-y.

Reply:
I tried throwing an error when the `owner` property is undefined, but the result of this was an empty autocomplete list. What do you think could be the issue and a possible solution?

Reply:
Is the error occurring in tests or when you live test it in the browser? You can walk me through in a VC